### PR TITLE
Update for raft logger changes

### PR DESCRIPTION
**Pull request by vyasr**
This PR updates cuvs to use raft's updated logger implementation using [rapids-logger](https://github.com/rapidsai/rapids-logger). It is a breaking change because it changes the kmeans `base_params` verbosity type from an int to a `raft::level_enum`.

This PR requires https://github.com/rapidsai/raft/pull/2530.

Contributes to https://github.com/rapidsai/build-planning/issues/104

### CODE REVIEW

**Comment by vyasr** [comment]
I'll be on vacation next week and this PR has downstream consequences so I've marked it as "DO NOT MERGE" until I get back, but I would love to get it reviewed in advance so that I can safely merge when I'm ready.

File: ci/build_cpp.sh
Comment (vyasr) [inline ci/build_cpp.sh:1]: Changes are for testing, will be removed before merge.

File: ci/build_python.sh
Comment (vyasr) [inline ci/build_python.sh:1]: Changes are for testing, will be removed before merge.

File: cpp/cmake/thirdparty/get_raft.cmake
Comment (vyasr) [inline cpp/cmake/thirdparty/get_raft.cmake:1]: Changes are for testing, will be removed before merge.

**Comment by vyasr** [comment]
The devcontainer failures are expected here until we merge the raft PR.

**Review by bdice — APPROVED** [review]

**Review by cjnolet — APPROVED** [review]

**Pull request merged**
